pandas/tools/merge.py

```diff
@@ -1626,7 +1626,11 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
                 clean_objs.append(v)
             objs = clean_objs
             name = getattr(keys, 'name', None)
-            keys = Index(clean_keys, name=name)
+            # GH 14848
+            # Don't pass name when creating index (# GH 14252)
+            # So that if keys are tuples, name isn't checked
```

Comment:
@jreback I'm not sure this can be fixed there, because the goal of the code in the section above was to fix an issue with the propagation of names in `Index` when using `concat()`. (Issue #14252). So what the fix I proposed does is allow the `Index` to be created, and then it copies the names.  If you consider the following code:
```
tl = [(0,0,1),(0,1,0),(1,0,0)]
i1 = pd.Index(tl, name='a')
i1
```
This code will raise an exception because it tries to compute a `MultiIndex` and there is only one name (single character `'a'`) supplied. But if the name argument is a 3 character string, i.e.,
```
i1 = pd.Index(tl, name='abc')
```
then the above snippet will work because the name is 3 characters long. In that case, it creates a `MultiIndex` with the names 'a', 'b' and 'c'.

I could modify `Index` to see if a single string is passed and not have that treated as a sequence for names for a `MultiIndex`, but I'd be concerned that would break code for others.